lib/parser/index.js

```diff
@@ -28,7 +28,8 @@ var $ = HTML.TAG_NAMES,
  */
 var DEFAULT_OPTIONS = {
     locationInfo: false,
-    treeAdapter: defaultTreeAdapter
+    treeAdapter: defaultTreeAdapter,
+    onToken: function () {}
```

Comment:
Initially, I actually hooked into private methods of `Parser` in the test suite, but then decided that it might be not the best idea and consumers might want to perform own actions on token retrieval as well (similarly to `treeAdapter`, but token-based). If you think exposing it is a bad idea and we should only use it for testing, then I think internal hook into `Parser` private methods would be better as it doesn't expose anything.